### PR TITLE
[UPD] g2p_registry_membership: monkey patching methods

### PR DESCRIPTION
**Pull request by nhatnm0612**
## What this PR does?

- [x] Create monkey patching to patch recompute methods for Odoo Environment.
- [x] Create test cases.

### CODE REVIEW

File: g2p_registry_membership/tests/test_indicator_recompute.py [hunk 1]
```diff
@@ -0,0 +1,150 @@
...
+            "Number of jobs should be the same since the number of records is 5!",
+        )
+        for partner in self._partners[:5]:
+            self.assertEqual(
+                partner.force_recompute_canary,
+                datetime(2023, 5, 18, 0, 0),
+                "force_recompute_canary should return mock time for now()!",
+            )
+
+    def test_04_compute_canary_more_than_10_partners(self):
```
Comment (jeremi) [inline g2p_registry_membership/tests/test_indicator_recompute.py:147]: Can you check that they are not returned when we call fields_to_compute?

File: g2p_registry_membership/tests/test_indicator_recompute.py [hunk 2]
```diff
@@ -0,0 +1,150 @@
...
+            "since the len of compute record is 11!",
+        )
+        self.assertTrue(
+            bool(self._partners - records_to_compute),
+            "Other fields recompute record should not include all the Canary Force Recompute field "
+            "since the len of compute record is 11!",
+        )
+
+    @patch("odoo.fields.Datetime.now")
+    def test_03_compute_canary_less_than_10_partners(self, mock_time):
```
Comment (jeremi) [inline g2p_registry_membership/tests/test_indicator_recompute.py:119]: Can you check that they are returned when we call fields_to_compute?

**Review by jeremi — CHANGES_REQUESTED** [review]

File: g2p_registry_membership/tests/test_indicator_recompute.py (hunk 2 above)
Comment (nhatnm0612) [reply g2p_registry_membership/tests/test_indicator_recompute.py:119]: sure

Comment (nhatnm0612) [reply g2p_registry_membership/tests/test_indicator_recompute.py:119]: done @jeremi

**Pull request closed without merging**